src/app/reducers/index.ts

```diff
@@ -21,6 +21,9 @@ const reducers = {
 };
 
 export function reducer(state: any, action: any) {
+  if (action && action.type === 'USER_LOGOUT') {
+    state = undefined;
+  }

```

Comment:
I chose to use [ngrx/store](https://github.com/ngrx/store) for app state management. This was actually that concept that I wanted to wrap my head around. Basically `state` is the object where all info about user's categories, items, and any other relevant app information would be stored. Setting it to undefined assures that other user, after logging in will not get access to previous user's info.